docs/myria-python/index.md

```diff
@@ -178,6 +178,98 @@ query = MyriaQuery.parallel_import(relation=relation, work=work)
 print query.status
 ```
 
+### Part 4. Python Expressions
+
+Myria allows arbitrary Python Functions as expressions which can be used in transformations like map, flatmap and aggregates.
+
+#### Registering a Python Function
+Python function needs to be registered before it can be used.
+Myria-Python can be used to register new Python functions, list existing registered Python functions,
+and to retrieve details about existing registered functions.
+
+```python
+from myria.udf import MyriaPythonFunction
+from raco.types import LONG_TYPE
+
+#create connection
+connection = MyriaConnection(rest_url='http://localhost:8753',execution_url='http://localhost:8080')
+
+#register Python functions
+outType= "LONG_TYPE"
+def pyIsPrime(dt):
+    import math
+    n = dt[0][0]
+    if n % 2 == 0 and n > 2: 
+        return False
+    for i in range(3, int(math.sqrt(n)) + 1, 2):
+        if n % i == 0:
+            return 0
+    return 1
+
+MyriaPythonFunction(pyIsPrime, outType).register()
+
+#List registered functions
+connection.get_functions()
+
+#List details of a registered function
+connection.get_function('pyIsPrime')
```

Comment:
myria-python has a higher-level API for getting functions (as entities, not JSON documents) as `MyriaFunction.get(name, connection)` and `MyriaFunction.get_all(connection)`.  There should be no for users to ever interact directly with the MyriaConnection facade.

I'll clean this up after I update the demo cluster.